src/cmd/services/m3query/config/config.go

```diff
@@ -283,6 +283,12 @@ type QueryConfiguration struct {
 	// RequireSeriesEndpointStartEndTime requires requests to /series endpoint
 	// to specify a start and end time to prevent unbounded queries.
 	RequireSeriesEndpointStartEndTime bool `yaml:"requireSeriesEndpointStartEndTime"`
+	// RewriteRangesLessThanResolutionMultiplier will rewrite the range in a query if it's
+	// determined that the namespaces used to service the request have resolution(s)
+	// that are greater than the range. The range will be updated to the largest resolution
+	// of the namespaces to service the request * the multiplier specified here. If this multiplier
+	// is 0, then this feature is disabled.
+	RewriteRangesLessThanResolutionMultiplier int `yaml:"rewriteRangesLessThanResolutionMultiplier"`
```

Comment:
This should live under `PrometheusQueryConfiguration` I believe since it's very Prometheus specific?

Probably `requireLabelsEndpointStartEndTime` and `requireSeriesEndpointStartEndTime` should too... but we can address that some other time perhaps.